### PR TITLE
Spatial_oM: Add support for tapered flange sections

### LINKED ISSUE #512: Spatial_oM: add tapered flange sections
To allow creation of sections such as:
C - Channels with tapered flanges, slope = 2/12
MC - Channels sometimes with tapered flanges, slope != 2/12
S - H Shapes with tapered flanges
M - H shapes which are not W or S shapes (sometimes have tapered flanges)

### PR DESCRIPTION
**Pull request by JosefTaylor**
### Issues addressed by this PR

Closes #512 

### Test files

https://burohappold.sharepoint.com/:f:/r/sites/BHoM/02_Current/12_Scripts/02_Pull%20Request/BHoM/BHoM/Spatial_oM/%231316-add-tapered-flange-sections?csf=1&web=1&e=aMhwGq

### CODE REVIEW

File: Spatial_oM/ShapeProfiles/TaperFlangeISectionProfile.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+        [Length]
+        public virtual double WebThickness { get; }
+
+        [Length]
+        [Description("Mean thickness of the flange, taken at 1/4 of the flange width.")]
+        public virtual double FlangeThickness { get; }
+
+        [Angle]
+        [Description("slope of the flange inner faces, generally around 0.16 radians.")]
```
Comment (IsakNaslundBh) [inline Spatial_oM/ShapeProfiles/TaperFlangeISectionProfile.cs:59]: Small suggestion of stating slightly larger range of commonly used values
```suggestion
        [Description("Slope of the flange inner faces, generally around 0.05 to 0.17 radians.")]
```

File: Spatial_oM/ShapeProfiles/TaperFlangeChannelProfile.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+        [Length]
+        public virtual double WebThickness { get; }
+
+        [Length]
+        [Description("Mean thickness of the flange, taken at 1/2 the flange width")]
+        public virtual double FlangeThickness { get; }
+
+        [Angle]
+        [Description("slope of the flange inner faces, generally around 0.16 radians.")]
```
Comment (IsakNaslundBh) [inline Spatial_oM/ShapeProfiles/TaperFlangeChannelProfile.cs:58]: Small suggestion of stating slightly larger range of commonly used values
```suggestion
        [Description("Slope of the flange inner faces, generally around 0.05 to 0.17 radians.")]
```

**Review by IsakNaslundBh — COMMENTED** [review]
Small suggestion on description

**Comment by JosefTaylor** [comment]
@BHoMBot check required

**Comment by JosefTaylor** [comment]
@BHoMBot check installer

**Comment by JosefTaylor** [comment]
@BHoMBot check required

**Comment by JosefTaylor** [comment]
@BHoMBot check code-compliance

**Comment by JosefTaylor** [comment]
@BHoMBot check required

**Comment by JosefTaylor** [comment]
This is failing because of some old, non-compliant code structure in Multi-Storey Toolkit, which I had to edit some references to Layer to disambiguate. I've requested dispensation over there, and am happy to raise an issue there as well.

**Comment by JosefTaylor** [comment]
Hmm, this doesn't make sense. None of the other ShapeProfile constructors take Shape as a parameter. Code-Compliance fails for me whether I set it in the constructor or not, as long as I don't have it as an input. We shouldn't be allowing folks to make channels with Shape set to ISection, so I don't want this in the constructor.

**Comment by IsakNaslundBh** [comment]
> Hmm, this doesn't make sense. None of the other ShapeProfile constructors take Shape as a parameter. Code-Compliance fails for me whether I set it in the constructor or not, as long as I don't have it as an input. We shouldn't be allowing folks to make channels with Shape set to ISection, so I don't want this in the constructor.

Agree. Shape type should not be provided in the constructor.

Need to check how to handle the compliance exception here.

**Review by IsakNaslundBh — APPROVED** [review]
looks good to me!

**Comment by IsakNaslundBh** [comment]
@BHoMBot check required

**Comment by IsakNaslundBh** [comment]
@BHoMBot this is a CI/CD instruction. I am authorising dispensation to be granted on check ref. 4301017643

**Comment by IsakNaslundBh** [comment]
@BHoMBot check copyright-compliance

**Comment by IsakNaslundBh** [comment]
@BHoMBot check ready-to-merge

**Comment by IsakNaslundBh** [comment]
Dispensating the code compliance check as this PR is following the exact same principles as all other shape profiles.

Both classes added have been tested for serialisation and both work fine. As this is the main reason for the failing compliance check I am comfortable with this.

**Pull request merged**